SparkPost/2web2ui
# FE-1147 - [Domains UX Redesign] Set up plumbing for the new feature

[FE-1147](https://jira.int.messagesystems.com/browse/FE-1147)

### What Changed
- Adds the following routes to the app behind the account UI option `allow_domains_v2`:
  - `/domains`
  - `/domains/create`
  - `/domains/:id`
  - `/domains/:id/verify-bounce`
  - `/domains/:id/verify-sending`
  - `/domains/:id/verify-tracking`
- Adds supporting Cypress tests
- Adds `DomainsLayout` component that will eventually provide access to the Redux store via a context-driven hook

### How To Test
- Enable Hibana and enable the account UI option `allow_domains_v2` and visit the aforementioned routes. Verify pages and titles render according to the mocks.
- Disable Hibana and try to visit the aforementioned routes and verify a redirect to the 404 page occurs
- Enable Hibana and disable the account UI option `allow_domains_V2` and visit the aforementioned routes. Verify you are redirected to the 404 page.

### To Do
